packages/hdwallet-vultisig/src/types.ts

```diff
@@ -0,0 +1,50 @@
+import { PublicKey, VersionedTransaction } from "@solana/web3.js";
+import { providers } from "ethers";
+
+import { BtcAccount } from "./bitcoin";
+import { SolanaAccount } from "./solana";
+
+type VultisigRequestParams = {
+  get_accounts: [];
+  request_accounts: [];
+};
+
+type VultisigRequestReturn = {
+  get_accounts: Promise<BtcAccount[] | string[]>;
+  request_accounts: Promise<BtcAccount[]>;
+};
+
+type VultisigRequestMethod = keyof VultisigRequestParams;
+
+type VultisigRequestPayload<M extends VultisigRequestMethod> = {
+  method: M;
+  params: VultisigRequestParams[M];
+};
+
+export type VultisigEvmProvider = providers.ExternalProvider;
+
+export type VultisigUtxoProvider = providers.ExternalProvider & {
+  request<M extends VultisigRequestMethod>(payload: VultisigRequestPayload<M>): Promise<VultisigRequestReturn[M]>;
+  signPSBT(
+    psbt: Uint8Array,
+    {
+      inputsToSign,
+    }: {
+      inputsToSign: {
+        address: string;
+        signingIndexes: number[];
+        sigHash?: number;
+      }[];
+    },
+    broadcast: boolean
+  ): Promise<Uint8Array>;
+};
+
+export type VultisigSolanaProvider = providers.ExternalProvider & {
+  publicKey?: PublicKey;
+  connect(): Promise<SolanaAccount>;
+  signTransaction(transaction: VersionedTransaction): Promise<VersionedTransaction>;
+  signAndSendTransaction(transaction: VersionedTransaction): Promise<{ signature: any }>;
```

Comment:
type `signAndSendTransaction` response

`import {  TransactionSignature } from "@solana/web3.js";`

```suggestion
  signAndSendTransaction(transaction: VersionedTransaction): Promise<{ signature: TransactionSignature }>;
```